#!/usr/share/ucs-test/runner python
## desc: Create DNS host record with wrong object type as superordinate
## tags: [udm-dns,apptest]
## roles: [domaincontroller_master]
## bugs: [15660]
## exposure: careful
## packages:
##   - univention-config
##   - univention-directory-manager-tools



import sys
import univention.testing.utils as util
import univention.testing.udm as udm_test
import univention.testing.strings as uts
import univention.uldap as uldap


if __name__ == '__main__':
	sys.exit(131) # skip for now

	with udm_test.UCSTestUDM() as udm:
		reverse_zone = udm.create_object('dns/reverse_zone', nameserver = uts.random_string(), subnet = '10.20.30')
		try:
			dns_host = udm.create_object('dns/host_record', name = uts.random_name(), superordinate = reverse_zone)
		except udm_test.UCSTestUDM_CreateUDMObjectFailed:
			sys.exit(0)

	

		ldap = uldap.getAdminConnection()
		ldap.delete(dns_host)
		utils.fail('UDM did not report an error while trying to create an host record with reverse zone as superordinate')
